lib/tilex_web/controllers/auth_controller.ex

```diff
@@ -32,21 +32,36 @@ defmodule TilexWeb.AuthController do
     |> redirect(to: "/")
   end
 
-  defp authenticate(%{info: info}) do
-    email = Map.get(info, :email)
-    name = Developer.format_username(Map.get(info, :name))
-
-    case String.match?(email, ~r/@#{Application.get_env(:tilex, :hosted_domain)}$/) do
-      true ->
+  defp authenticate(%{info: %{email: email, name: name}}) when is_binary(name) do
+    case authorized(email) do
+      {:ok, email} ->
         attrs = %{
           email: email,
-          username: name
+          username: Developer.format_username(name)
         }
 
         Developer.find_or_create(Repo, attrs)
 
       _ ->
-        {:error, email}
+        {:error, "#{email} is not a valid email address"}
+    end
+  end
+
+  defp authenticate(_), do: {:error, "oauth2 profile is missing a valid name"}
+
+  defp authorized(email) do
+    cond do
+      String.match?(email, ~r/@#{hosted_domain()}$/) -> {:ok, email}
+      email in guest_whitelist() -> {:ok, email}
+      true -> {:error, email}
     end
   end
+
+  defp hosted_domain, do: Application.get_env(:tilex, :hosted_domain)
+
+  defp guest_whitelist do
+    :tilex
+    |> Application.get_env(:guest_author_whitelist)
+    |> String.split(", ")
```

Comment:
Guess we could make the delimiter configurable.

Reply:
I think it's OK to not make it configurable, but prefer if it was like this:

```elixir
"hello, hey, hi, dude " |> String.split([",", " "], trim: true)
# => ["hello", "hey", "hi", "dude"]
```
This way if they don't put the spaces after the comma it will still work:

```elixir
"hello,hey,hi, dude" |> String.split([",", " "], trim: true)
# => ["hello", "hey", "hi", "dude"]
```

Reply:
Ah nice didn't know you could do this!